Content.Server/Medical/SuitSensors/SuitSensorSystem.cs

```diff
@@ -0,0 +1,291 @@
+using Robust.Shared.Localization;
+using Robust.Shared.Map;
+using Robust.Shared.Player;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.Medical.SuitSensors
+{
+    public class SuitSensorSystem : EntitySystem
+    {
+        [Dependency] private readonly ActionBlockerSystem _actionBlockerSystem = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly IRobustRandom _random = default!;
+        [Dependency] private readonly IdCardSystem _idCardSystem = default!;
+        [Dependency] private readonly DeviceNetworkSystem _deviceNetworkSystem = default!;
+        [Dependency] private readonly IGameTiming _gameTiming = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<SuitSensorComponent, MapInitEvent>(OnMapInit);
+            SubscribeLocalEvent<SuitSensorComponent, EquippedEvent>(OnEquipped);
+            SubscribeLocalEvent<SuitSensorComponent, UnequippedEvent>(OnUnequipped);
+            SubscribeLocalEvent<SuitSensorComponent, ExaminedEvent>(OnExamine);
+            SubscribeLocalEvent<SuitSensorComponent, GetInteractionVerbsEvent>(OnVerb);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            var sensors = EntityManager.EntityQuery<SuitSensorComponent, DeviceNetworkComponent>();
+            foreach (var (sensor, device) in sensors)
+            {
+                // check if sensor is ready to update
+                if (_gameTiming.CurTime - sensor.LastUpdate < TimeSpan.FromSeconds(sensor.UpdateRate))
+                    continue;
+                sensor.LastUpdate = _gameTiming.CurTime;
+
+                // get sensor status
+                var status = GetSensorState(sensor.OwnerUid, sensor);
+                if (status == null)
+                    continue;
+
+                // broadcast it to device network
+                var payload = SuitSensorToPackage(status);
+                _deviceNetworkSystem.QueuePacket(sensor.OwnerUid, DeviceNetworkConstants.NullAddress, device.Frequency, payload, true);
+            }
+        }
+
+        private void OnMapInit(EntityUid uid, SuitSensorComponent component, MapInitEvent args)
+        {
+            // generate random mode
+            if (component.RandomMode)
+            {
+                //make the sensor mode favor higher levels, except coords.
+                var modesDist = new[]
+                {
+                    SuitSensorMode.SensorOff,
+                    SuitSensorMode.SensorBinary, SuitSensorMode.SensorBinary,
+                    SuitSensorMode.SensorVitals, SuitSensorMode.SensorVitals, SuitSensorMode.SensorVitals,
+                    SuitSensorMode.SensorCords, SuitSensorMode.SensorCords
+                };
+                component.Mode = _random.Pick(modesDist);
+            }
```

Comment:
this should be done on componentinit, not mapinit imo

Reply:
what the difference?

Reply:
both work, but i'd still rather have it on compinit since its not maprelated.

compinit -> when the component gets intialized on entitycreation
mapinit -> when the map the entity is on gets intialized

Reply:
> both work, but i'd still rather have it on compinit since its not maprelated.

But why? Right now if you save and load map - suit sensor will save its mode. But if you move it to init, it will be rewriting its state with each reload.

Reply:
afaict mapinit should still run, even if you load a map

Reply:
the idea is to have this run everytime a component is intialized, not everytime the map the components entity is on get initialized, since the first makes more sense